src/app/components/account/account-stacks-address.tsx

```diff
@@ -0,0 +1,12 @@
+import { Caption } from '@leather.io/ui';
+import { truncateMiddle } from '@leather.io/utils';
+
+import { useStacksAccountLoader } from '../loaders/stacks-account-loader';
+
+interface AccountStacksAddressProps {
+  index: number;
+}
+export function AccountStacksAddress({ index }: AccountStacksAddressProps) {
+  const account = useStacksAccountLoader({ index });
+  return account ? <Caption>{truncateMiddle(account.address, 4)}</Caption> : null;
+}
```

Comment:
If this component reads app state, it should really be a feature, even if a very small one? What do you think?

Reply:
Also, what happens if user wallet only has bitcoin keys?

Reply:
This is using an account loader for Stacks, so I think ok. It is used the same in `account-addresses` where both are used together. I will double check though and make sure.

I will relocate all the address components bc they are using state, good point. I'll make sure we can't pass in state where they are used.